plugins/extractors/tableau/models.go

```diff
@@ -0,0 +1,196 @@
+	source := mapConnectionTypeToSource(dbs.ConnectionType)
+
+	var urn string
+	switch source {
+	case "bigquery":
+		// bigquery::sample-project/dataset_a/invoice
+		// sometimes table name can be the same as full name (e.g. project-id.schema.table1), so we build URN with the full name instead
+		fullNameSplitted, err := parseBQTableFullName(tableInfo.FullName)
+		if err != nil {
+			// assume fullNameSplitted[0] is the project ID
+			urn = models.TableURN(source, fullNameSplitted[0], tableInfo.Schema, tableInfo.Name)
+			break
+		}
+		urn = models.TableURN(source, fullNameSplitted[0], fullNameSplitted[1], fullNameSplitted[2])
+	default:
+		// postgres::postgres:5432/postgres/user
+		host := fmt.Sprintf("%s:%d", dbs.HostName, dbs.Port)
+		urn = models.TableURN(source, host, dbs.Name, tableInfo.Name)
+
+	}
+	resource = &common.Resource{
+		Urn:     urn,
+		Type:    "table",
+		Service: source,
+	}
+	return
+}
+
+type CloudFile struct {
+	ID             string `json:"id"`
+	Name           string `json:"name"`
+	ConnectionType string `json:"connectionType"`
+	Description    string `json:"description"`
+	Provider       string `json:"provider"`
+	FileExtension  int    `json:"fileExtension"`
+	MimeType       string `json:"mimeType"`
+	RequestURL     string `json:"requestUrl"`
+}
+
+func (cf *CloudFile) CreateResource(tableInfo Table) (resource *common.Resource) {
+	source := mapConnectionTypeToSource(cf.ConnectionType)
+	urn := fmt.Sprintf("%s::%s/%s/%s", source, cf.Provider, cf.Name, tableInfo.Name)
+	resource = &common.Resource{
+		Urn:     urn,
+		Type:    "bucket", // TODO need to check what would be the appropriate type for this
+		Service: source,
+	}
+	return
+}
+
+type File struct {
+	ID             string `json:"id"`
+	Name           string `json:"name"`
+	ConnectionType string `json:"connectionType"`
+	Description    string `json:"description"`
+	FilePath       string `json:"filePath"`
+}
+
+func (f *File) CreateResource(tableInfo Table) (resource *common.Resource) {
+	source := mapConnectionTypeToSource(f.ConnectionType)
+	urn := fmt.Sprintf("%s::%s/%s/%s", source, f.FilePath, f.Name, tableInfo.Name)
+	resource = &common.Resource{
+		Urn:     urn,
+		Type:    "bucket", // TODO need to check what would be the appropriate type for this
+		Service: source,
+	}
+	return
+}
+
+type WebDataConnector struct {
+	ID             string `json:"id"`
+	Name           string `json:"name"`
+	ConnectionType string `json:"connectionType"`
+	Description    string `json:"description"`
+	ConnectorURL   string `json:"connectorUrl"`
+}
+
+func (wdc *WebDataConnector) CreateResource(tableInfo Table) (resource *common.Resource) {
+	source := mapConnectionTypeToSource(wdc.ConnectionType)
+	urn := fmt.Sprintf("%s::%s/%s/%s", source, wdc.ConnectorURL, wdc.Name, tableInfo.Name)
+	resource = &common.Resource{
+		Urn:     urn,
+		Type:    "table", // TODO need to check what would be the appropriate type for this
```

Comment:
@StewartJingga need your confirmation on this `type`, what would be the most suitable `type` for this resource?